### PR TITLE
Feature: Support for multi value variable queries

### PR DESCRIPTION
**Pull request by yesoreyeram**
This PR provides support for #90, #112, #113 via `$__multiVar()` macro in query fields. 

https://user-images.githubusercontent.com/153843/164561919-414183c3-c57c-4c7c-b400-bf1bb5043668.mov

### CODE REVIEW

File: src/variables.ts [hunk 1]
```diff
@@ -2,22 +2,35 @@ import { GitHubQuery } from './types';
...
+      }
     }
   });
 
   if (query.options) {
     let { options } = query;
     Object.keys(options).forEach((key) => {
       if (typeof options[key] === 'string') {
-        options = { ...options, [key]: replaceVariable(t, options[key], scoped) };
+        if (key === 'query') {
```
Comment (scottlepp) [inline src/variables.ts:29]: looks like we are repeating ourselves here.  same as https://github.com/grafana/github-datasource/pull/162/files#diff-7b3e863a7f8b31310b0cd25e6933ac6ae774750e4399e9621f11af435172d2d2R17-R21

maybe create a function to DRY it up

File: pkg/github/macros.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+			out := ""
+			prop := ""
+			for idx, arg := range args {
+				if idx == 0 {
+					prop = arg
+					continue
+				}
+				out = strings.Trim(fmt.Sprintf("%s %s:%s", out, prop, arg), " ")
+			}
+			if len(args) <= 1 {
```
Comment (scottlepp) [inline pkg/github/macros.go:42]: if we favor early returns, we would move these validations to the top before trying to loop over the args

**Review by scottlepp — APPROVED** [review]
LGTM - some minor suggestions

File: src/variables.ts (hunk 1 above)
Comment (yesoreyeram) [reply src/variables.ts:28]: 👍🏼 addressed in https://github.com/grafana/github-datasource/pull/162/commits/7c2a17d400fa07c4ad88e25019e234e1f2d28954 and added tests.

We should also aim https://github.com/grafana/github-datasource/issues/43 if we decide to refactor this further.

File: pkg/github/macros.go (hunk 2 above)
Comment (yesoreyeram) [reply pkg/github/macros.go]: 👍🏼 addressed in https://github.com/grafana/github-datasource/pull/162/commits/66119dfb4d970c02392590d087f63dfb1b49360f

**Comment by CodyGirl** [comment]
Hi @yesoreyeram , Does this feature also work with github enterprise URL? What should be passed as prefix.. in your example video clip it is 'repo' but I'm confused if it is just a placeholder or is it the variable 'repo' you've on the dashboard?
For my use case the variable name is $repositories... should I call the query as $__multiVar(repo,$repositories) (Tried with github enterprise but didn't work)?

**Comment by CodyGirl** [comment]
Thank you for your quick response. However, it is not working for me, here is a screenshot for the same:
image
I used Query inspector to see the query and the goes something like this: $__multiVar(repo,repo1,repo2...). Please help me if I'm doing something wrong.

**Comment by CodyGirl** [comment]
Any comments on this? It'll really be helpful to have your inputs.

**Pull request merged**